### PR TITLE
Update dev bits

### PR DESCRIPTION
**Pull request by weshooper**
- Update dev dependencies (and a few minor tweaks to accommodate)
- Add required PHP extensions
- Configure CI to build with 7.2 and 7.3
- Remove redundant fields in `composer.json`

### CODE REVIEW

File: tests/XmlTest.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 namespace DemandwareXml\Test;
 
 use DemandwareXml\Xml;
+use DemandwareXml\XmlException;
```
Comment (WillJW) [inline tests/XmlTest.php]: This can be merged with the include above

File: tests/ParserTest.php [hunk 2]
```diff
@@ -2,19 +2,9 @@
...
-    CategorySimpleNodeParser,
-    ProductNodeParser,
-    ProductSimpleNodeParser,
-    SetNodeParser,
-    SetSimpleNodeParser,
-    VariationNodeParser,
-    VariationSimpleNodeParser
-};
+use DemandwareXml\Parser\{AssignmentNodeParser, BundleNodeParser, BundleSimpleNodeParser, CategoryNodeParser, CategorySimpleNodeParser, ProductNodeParser, ProductSimpleNodeParser, SetNodeParser, SetSimpleNodeParser, VariationNodeParser, VariationSimpleNodeParser};
+use DemandwareXml\XmlException;
```
Comment (WillJW) [inline tests/ParserTest.php]: This can be merged with `DemandwareXml\Parser` above.

File: .semaphore/semaphore.yml [hunk 3]
```diff
@@ -13,10 +13,13 @@ blocks:
...
       prologue:
         commands:
-          - sem-version php 7.2.10 # Be explicit, even when using the default (in case the default changes)
+          - sem-version php $PHP_VERSION
           - checkout
       jobs:
       - name: Composer install
+        matrix:
+          - env_var: PHP_VERSION
+            values: ["7.2.10", "7.3.0"] # Be explicit, even when using the default (in case the default changes)
```
Comment (WillJW) [inline .semaphore/semaphore.yml]: What does it mean to have two values?

Comment (WillJW) [reply .semaphore/semaphore.yml]: Never mind, I worked it out 😄

File: tests/ParserTest.php (hunk 2 above)
Comment (weshooper) [reply tests/ParserTest.php]: Done, thanks @WillJW

**Pull request merged**